src/categorical_algebra/CSets.jl

```diff
@@ -1439,6 +1447,13 @@ function hom(A::SubACSetComponentwise{T}) where T <: ACSet
   U = constructor(X)()
   hom_components = map(collect∘hom, components(A))
   copy_parts!(U, X, hom_components)
+  # Reindex attrvars in the domain
```

Comment:
Does `copy_parts!` not support attribute variables? It seems like this logic shouldn't need to be here.

Reply:
`copy_parts!` does copy attribute variables, but does so literally (it doesn't reindex them).

You're right that it should probably also handle the reindexing logic, but that would be a change to ACSets.jl.

Reply:
OK, yes, that does sound like a change that should be made to ACSets. Can you create an issue upstream?

Reply:
https://github.com/AlgebraicJulia/ACSets.jl/issues/38